src/test/cypress/integration/exam/ExamAssessment.spec.ts

```diff
@@ -0,0 +1,184 @@
+                cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+                examStartEnd.startExam();
+                cy.contains(modelingResponse.body.title).should('be.visible').click();
+                modelingEditor.addComponentToModel(1);
+                modelingEditor.addComponentToModel(2);
+                modelingEditor.addComponentToModel(3);
+                cy.intercept(PUT, BASE_API + 'exercises/*/modeling-submissions').as('createModelingSubmission');
+                cy.contains('Save').click();
+                cy.wait('@createModelingSubmission');
+                cy.get('#exam-navigation-bar').find('.btn-danger').click();
+                examStartEnd.finishExam();
+                cy.login(tutor, '/course-management/' + course.id + '/exams');
+                cy.contains('Assessment Dashboard', { timeout: 60000 }).click();
+                assessmentDashboard.startAssessing();
+                modelingAssessment.addNewFeedback(2, 'Noice');
+                modelingAssessment.openAssessmentForComponent(1);
+                modelingAssessment.assessComponent(1, 'Good');
+                modelingAssessment.openAssessmentForComponent(2);
+                modelingAssessment.assessComponent(0, 'Neutral');
+                modelingAssessment.openAssessmentForComponent(3);
+                modelingAssessment.assessComponent(-1, 'Wrong');
+                assessmentDashboard.submitModelingAssessment().then((assessmentResponse) => {
+                    expect(assessmentResponse.response?.statusCode).to.equal(200);
+                });
+                cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+                cy.contains('2 of 10 points').should('be.visible');
+            });
+        });
+
+        it('Assess a text exercise submission', () => {
+            const exerciseTitle = 'Cypress Text Exercise';
+            courseManagementRequests.createTextExercise({ exerciseGroup }, exerciseTitle);
+            courseManagementRequests.generateMissingIndividualExams(exam);
+            courseManagementRequests.prepareExerciseStartForExam(exam);
+            cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+            examStartEnd.startExam();
+            cy.contains(exerciseTitle).click();
+            cy.get('#text-editor-tab').type(textSubmission.text);
+            cy.contains('Save').click();
+            cy.get('#exam-navigation-bar').find('.btn-danger').click();
+            examStartEnd.finishExam();
+            cy.login(tutor, '/course-management/' + course.id + '/exams');
+            cy.contains('Assessment Dashboard', { timeout: 60000 }).click();
+            assessmentDashboard.startAssessing();
+            assessmentDashboard.addNewFeedback(7, 'Good job');
+            assessmentDashboard.submitTextAssessment().then((assessmentResponse) => {
+                expect(assessmentResponse.response?.statusCode).to.equal(200);
```

Comment:
```suggestion
                expect(assessmentResponse.response.statusCode).to.equal(200);
```

Reply:
this is a bad suggestion. Typescript will give a 'Object possibly undefined' which we could ignore semantically but then the code is more ugly than with this ?.